File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def fun(self,s,low,high):

        # non-alphanumeric characters ko skip karo
        while low<high and not s[low].isalnum():
            low+=1

        while low<high and not s[high].isalnum():
            high-=1

        # Base Case
        if low>=high:
            return True

        # agar first aur last character equal nahi hai
        if s[low].lower()!=s[high].lower():
            return False

        # Recursive Call nxt walo k liye
        return self.fun(s,low+1,high-1)

    def isPalindrome(self,s):

        # Size of the String
        n=len(s)

        low=0
        high=n-1

        # Recursive Fn. call kardo
        return self.fun(s,low,high)

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool fun(string& s,int low,int high){

        // non-alphanumeric characters ko skip karo
        while(low<high && !isalnum(s[low])){
            low++;
        }

        while(low<high && !isalnum(s[high])){
            high--;
        }

        // Base Case
        if(low>=high){
            return true;
        }

        // agar first aur last character equal nahi hai
        if(tolower(s[low])!=tolower(s[high])){
            return false;
        }

        // Recursive Call nxt walo k liye
        return fun(s,low+1,high-1);
    }

    bool isPalindrome(string s){

        // Size of the String
        int n=s.size();

        int low=0;
        int high=n-1;

        // Recursive Fn. call kardo
        return fun(s,low,high);
    }
};